### PR TITLE
NotificationHubInstallationAdapter should include templates in body

### PR DESCRIPTION
**Pull request by olegErmakov**

### CODE REVIEW

File: notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/InstallationTemplate.java [hunk 1]
```diff
@@ -97,6 +102,69 @@ public void removeHeader(String name) {
...
+
+    /**
+     * Deserialize JSONObject to InstallationTemplate.
+     *
+     * @param installationTemplate Template to deserialize
+     * @return Deserialized template.
+     * @throws JSONException When there's a schema-mismatch of the object to populate, and what
+     * appears in the serialized form of the template.
+     */
+    public static InstallationTemplate deserialize(JSONObject installationTemplate) throws JSONException {
```
Comment (marstr) [inline notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/InstallationTemplate.java:145]: Same here, let's make it package-private.

File: notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/InstallationTemplate.java [hunk 2]
```diff
@@ -97,6 +102,69 @@ public void removeHeader(String name) {
...
     }
 
+    /**
+     * Serialize InstallationTemplate to JSONObject.
+     *
+     * @param name Name of template
+     * @param installationTemplate The templates that should no longer be in the collection.
+     * @return serialized templateObject.
+     */
+    public static JSONObject serialize(String name, InstallationTemplate installationTemplate) {
```
Comment (marstr) [inline notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/InstallationTemplate.java:112]: Let's make this package-private. I don't want to leak public surface area. :)

Part of it too is that at the moment there's only the one JSON serialization form that is important, but in the future we could conceivably need to serialize this differently for different scenarios. For instance, API-Versions could require different formats, or we could even want to store the template differently in Shared preferences than we do when it hits the wire.

Comment (iTOYS) [reply notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/InstallationTemplate.java]: fixed

File: notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/InstallationTemplate.java (hunk 1 above)
Comment (iTOYS) [reply notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/InstallationTemplate.java]: fixed

**Pull request merged**